code/controllers/subsystem/statpanel.dm

```diff
@@ -395,11 +395,62 @@ SUBSYSTEM_DEF(statpanels)
 	listed_turf = new_turf
 	if(!client)
 		return
-	if(!client.obj_window)
-		client.obj_window = new(client)
-	if(listed_turf)
-		client.stat_panel.send_message("create_listedturf", listed_turf.name)
-		client.obj_window.start_turf_tracking()
-	else
-		client.stat_panel.send_message("remove_listedturf")
-		client.obj_window.stop_turf_tracking()
+
+	var/datum/tgui/ui = SStgui.try_update_ui(usr, src, "LootPanel")
+	if(!ui)
+		ui = new(usr, src, "LootPanel")
+		ui.open()
+
+/mob/ui_data(mob/user)
+	var/list/data = list()
+
+	var/list/contents = list()
+	for(var/atom/thing in listed_turf.contents)
+		var/generated_string
+		if(ismob(thing) || length(thing.overlays) > 2)
+			generated_string = costly_icon2html(thing, client, sourceonly=TRUE)
```

Comment:
I created a caching and queuing system for this with stat panels, please use it
Something similar to ensure we update smoothing instead of on a delay https://github.com/tgstation/tgstation/pull/69825

Reply:
we should find a way to support this kind of thing in a 514/515 compatible manner

Reply:
If it updates at all I want to. I was still trying to grok how your system worked, but just wanted the window to work first